rpcperms/middleware_handler.go

```diff
@@ -529,3 +530,43 @@ func parseProto(typeName string, serialized []byte) (proto.Message, error) {
 
 	return msg.Interface(), nil
 }
+
+// replaceProtoMsg replaces the given target message with the content of the
```

Comment:
I need to dig more into the proto internals for the Go library, but is there a reason we can't just literally swap out these pointers?

Reply:
Unfortunately pointer magic doesn't work here, because we have a generic interface type. We can't do `*ptr = newStruct{}` unless we know exactly what type it is. Which would make this non-generic.